File: java/snippets/configuration/InitDefaultConfig.java
package configuration;

import com.senzing.sdk.*;
import com.senzing.sdk.core.SzCoreEnvironment;

/**
 * Provides a simple example of adding records to the Senzing repository.
 */
public class InitDefaultConfig {
    public static void main(String[] args) {
        // get the senzing repository settings
        String settings = System.getenv("SENZING_ENGINE_CONFIGURATION_JSON");
        if (settings == null) {
            System.err.println("Unable to get settings.");
            throw new IllegalArgumentException("Unable to get settings");
        }

        // create a descriptive instance name (can be anything)
        String instanceName = InitDefaultConfig.class.getSimpleName();

        // initialize the Senzing environment
        SzEnvironment env = SzCoreEnvironment.newBuilder()
            .settings(settings)
            .instanceName(instanceName)
            .verboseLogging(false)
            .build();
        
        try {
            // get the config and config manager from the environment
            SzConfigManager configMgr = env.getConfigManager();

            // prepare a config to be modified
            SzConfig    config              = configMgr.createConfig();
            String      configDefinition    = config.export();

            // register the modified config in the repository as the default
            configMgr.setDefaultConfig(configDefinition);

        } catch (SzException e) {
            // handle any exception that may have occurred
            System.err.println("Senzing Error Message : " + e.getMessage());
            System.err.println("Senzing Error Code    : " + e.getErrorCode());
            e.printStackTrace();
            throw new RuntimeException(e);

        } catch (Exception e) {
            e.printStackTrace();
            if (e instanceof RuntimeException) {
                throw ((RuntimeException) e);
            }
            throw new RuntimeException(e);

        } finally {
            // IMPORTANT: make sure to destroy the environment
            env.destroy();
        }

    }
}

File: python/configuration/init_default_config.py
#! /usr/bin/env python3

import os
import sys
from pathlib import Path

from senzing import SzError
from senzing_core import SzAbstractFactoryCore

EXISTING_CONFIG_MSG = "\nA configuration exists in the repository, replace it with a template configuration? (y/n)  "
INSTANCE_NAME = Path(__file__).stem
SETTINGS = os.getenv("SENZING_ENGINE_CONFIGURATION_JSON", "{}")


try:
    sz_factory = SzAbstractFactoryCore(INSTANCE_NAME, SETTINGS, verbose_logging=False)
    sz_configmanager = sz_factory.create_configmanager()

    if current_config_id := sz_configmanager.get_default_config_id():
        if not input(EXISTING_CONFIG_MSG).lower() in ("y", "yes"):
            sys.exit(1)

    sz_config = sz_configmanager.create_config_from_template()
    new_default_config = sz_config.export()
    new_config_id = sz_configmanager.set_default_config(new_default_config, "Code snippet init_default_config example")
    print(f"New default config ID: {new_config_id}")
except SzError as err:
    print(f"{err.__class__.__name__} - {err}", file=sys.stderr)
